engine/src/main/java/org/hibernate/search/engine/metadata/impl/PropertyMetadata.java

```diff
@@ -64,11 +64,18 @@ public String getPropertyAccessorName() {
 		return propertyAccessorName;
 	}
 
-	// TODO HSEARCH-1867 change return type to set
+	/**
+	 * @deprecated use getFieldMetadataSet() instead
+	 */
+	@Deprecated
 	public List<DocumentFieldMetadata> getFieldMetadata() {
 		return new ArrayList<DocumentFieldMetadata>( documentFieldMetadataList );
 	}
 
+	public Set<DocumentFieldMetadata> getFieldMetadataSet() {
```

Comment:
I don't remember the details, but the fact that we had a full JIRA for it rather than having it fixed in the past makes me unformfortable in fixing it in a Final micro release. This needs to be backported as it's a performance issue.